configure.ac

```
@@ -644,12 +644,16 @@ AC_TRY_COMPILE([#include <sys/time.h>],
                [struct timeval *tv;
                 struct timezone *tz;],
                AC_DEFINE(HAVE_STRUCT_TIMEVAL,1,[Define if we have a timeval structure])
-               AC_TRY_COMPILE([#include <sys/time.h>],
-                              [struct timeval *tv;
-                               struct timezone *tz;
-                               gettimeofday(tv, tz);],
-                              AC_DEFINE(HAVE_GETTIMEOFDAY,1,[Define if gettimeofday is found]),
-                              AC_DEFINE(NEED_GETTIMEOFDAY,1,[Define if gettimeofday is needed])))
+			   FOUND_STRUCT_TIMEVAL="yes")
+
+if test x$FOUND_STRUCT_TIMEVAL = x"yes"; then
```

```suggestion
if test "$FOUND_STRUCT_TIMEVAL" = yes; then
```

---

Ähm ... good catch, look wired.

---

Anyway ... https://github.com/monitoring-plugins/monitoring-plugins/pull/1927/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810L630 looks equal

---

isn't that how one has to compare strings to be the comparison does the right thing if one of the strings is empty?

---

I'd opt for

```suggestion
if test x"$FOUND_STRUCT_TIMEVAL" = x"yes"; then
```

or

```suggestion
if test "x$FOUND_STRUCT_TIMEVAL" = "xyes"; then
```